examples/generateCatalog.py

```diff
@@ -0,0 +1,183 @@
+
+	### Write header of instance catalog file with observing and simulation parameters
+        outfile = open(outfilename, "w")
+	print >> outfile, 'Unrefracted_RA', ra_cen
+	print >> outfile, 'Unrefracted_Dec', dec_cen
+	print >> outfile, 'Opsim_obshistid %d'%histid
+	print >> outfile, 'SIM_SEED %d'%seed
+	print >> outfile, 'SIM_EXPTIME 30.0'
+	print >> outfile, 'Unrefracted_RA 81.1760879'
+	print >> outfile, 'Unrefracted_Dec -12.2103036'
+	print >> outfile, 'Opsim_moonra 313.898938'
+	print >> outfile, 'Opsim_moondec -12.7605726'
+	print >> outfile, 'Opsim_rotskypos 184.222551'
+	print >> outfile, 'Opsim_rottelpos 0'
+	print >> outfile, 'Opsim_filter %d'%filter
+	print >> outfile, 'Opsim_rawseeing %f'%seeing
+	print >> outfile, 'Opsim_sunalt -18.1759974'
+	print >> outfile, 'Opsim_moonalt -25.2607921'
+	print >> outfile, 'Opsim_dist2moon 122.048036'
+	print >> outfile, 'Opsim_moonphase 0.394149005'
+	print >> outfile, 'Opsim_expmjd 50486.0469'
+	print >> outfile, 'Opsim_altitude 72.4911951'
+	print >> outfile, 'Opsim_azimuth 355.24931'
+	print >> outfile, 'SIM_VISTIME 30.0'
+	print >> outfile, 'SIM_NSNAP 1'
+
+	### Write sources to instance catalog file
+	### Columns in the instance catalog:
+	###   ["object", "id", "ra", "dec", "mag_norm", "sed_name",
+	###	   "redshift", "gamma1", "gamma2", "kappa", "delta_ra", "delta_dec",
+	###	   "source_type", "dust_rest_name", "dust_lab_name"]
+	for i in xrange(len(ra)):
+		print >> outfile, 'object', str(i), str(ra[i]), str(dec[i]), '17.5', sed_filename, '0 0 0 0 0 0 star none none'
+	outfile.close()
+
+	# stars = pd.DataFrame({"object": "object",
+	#	"id": np.arange(nstars_per_dim * nstars_per_dim, dtype=float),
+	#	"ra": ra.ravel(),
+	#	"dec": dec.ravel(),
+	#	"mag_norm": 30.,
+	#	"sed_name": sed_filename,
+	#	"redshift": 0.0,
+	#	"gamma1": 0,
+	#	"gamma2": 0,
+	#	"kappa": 0,
+	#	"delta_ra": 0,
+	#	"delta_dec": 0,
+	#	"source_type": "star",
+	#	"dust_rest_name": "none",
+	#	"dust_lab_name": "none"},
+	#	columns=["object", "id", "ra", "dec", "mag_norm", "sed_name",
+	#			 "redshift", "gamma1", "gamma2", "kappa", "delta_ra", "delta_dec",
+	#			 "source_type", "dust_rest_name", "dust_lab_name"])
+	# stars.to_csv(outfile, sep=" ", header=False, index=False)
+	return None
+
+# def run_phosim():
+# 	# And run phosim over this new file. Note that we've specified
+# 	# no sky background, to speed things up.
+# 	subcmd = ['python', phosim_path+"phosim.py", catfilename,  "-c", phosim_path+"examples/nobackground", "-s", sensor, "-w", workdir, "-o", outdir]
+
+# 	subprocess.call(subcmd)
+
+
+###########################################################
+# Test functions for this script
+###########################################################
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='Phosim input file generator.')
+    parser.add_argument('--outfile', default="star_grid.pars", help='Output file')
+    parser.add_argument('--nstars_per_dim', type=int, default=100,
```

Comment:
Sorry for the duplication again.  I just want to make sure you see it.  I think density is probably a better argument than number.  It's just more intuitive and it doesn't have to change if you change the size of the field.